dvc/stage.py

```diff
@@ -498,8 +498,6 @@ def create(repo, **kwargs):
             stage.remove_outs(ignore_remove=False)
             logger.warning("Build cache is ignored when using --remove-outs.")
             ignore_build_cache = True
-        else:
-            stage.unprotect_outs()
```

Comment:
Why is this removed?

Reply:
So, there used to be an option for dvc commands, `--remove-outs` (that is now deprecated and default behaviour) that would remove existing outs before running command. Without specifying this option, if file was protected, we were unable to write to it. This part of code is leftover from when `--remove-outs` needed to be specified. When  removing became default behaviour this piece of code should have been removed. Unprotecting is now required only in case of persistent outputs, but `remove_outs` takes care of that (btw, it should probably be named different, since outs are not removed in some cases)